_rules/aria-state-or-property-permitted-5c01ea.md

```diff
@@ -58,6 +58,8 @@ Implementation of [Presentational Roles Conflict Resolution][] varies from one b
 
 In HTML, there are language features that do not have corresponding implicit WAI-ARIA semantics. As per [ARIA in HTML](https://www.w3.org/TR/html-aria/), those elements can have [global states or properties][global]. Some of those elements can also have [inherited][], [supported][], or [required][] [states][state] or [properties][property] that correspond to a [WAI-ARIA role](https://www.w3.org/TR/wai-aria-1.2/#introroles). For example, the `audio` element has no corresponding ARIA semantics but it can have [inherited][], [supported][], or [required][] [states][state] or [properties][property] of the [`application` role](https://www.w3.org/TR/wai-aria-1.2/#application).
 
+As per the [conflicts with host language semantics](https://www.w3.org/TR/wai-aria-1.2/#host_general_conflict) section of [WAI-ARIA 1.2](https://www.w3.org/TR/wai-aria-1.2/), user agents must ignore [WAI-ARIA state or property][] that have the same [implicit WAI-ARIA semantic](https://www.w3.org/TR/wai-aria-1.2/#implicit_semantics) in order to avoid providing conflicting states and properties to assistive technologies.
+
```

Comment:
I'm not sure this (and new example) is actually an ARIA problem 🤔 

---
The WAI-ARIA section you reference mostly mention user agents behaviour, and host languages specs. It seems the only part that refers authors behaviour is:
> When a feature in the host language with identical role semantics and values is available, and the author has no compelling reason to avoid using the host language feature, authors SHOULD use the host language features rather than repurpose other elements with WAI-ARIA.

but then the "SHOULD" bit make it not mandatory.
So I do feel that ARIA does not consider `aria-checked` as prohibited when `checked` is here.

Similarly, HTML AAM for `checked` states:
> If the element includes both the checked attribute and the aria-checked attribute with a valid value, User Agents MUST expose only the checked attribute value.

this corresponds to the WAI-ARIA bit "host languages MUST explicitly declare where the use of WAI-ARIA attributes on each host language element conflicts with native attributes for that element". But again, this (and the paragraph this comes from) is about host language and UA requirements, not author requirements.

So, I feel it does have some value to flag it (as in "it won't act as you expect"), but this is more of a Best Practice than really a requirement. Notably, `<input type="checkbox" aria-checked="mixed" />` won't cause any accessibility problem per se with conforming UAs that are just going to discard the `aria-checked` (it may cause some if the author relies on `aria-checked` somewhere else, but that checkbox alone is OK.

---
Would it make sense to keep that out of this rule and have another rule "conflicting property is not used"?

Reply:
Hey @Jym77 
I used the quote from ARIA as it seems to explain the issue well. I should've probably used ARIA in HTML to make it clearer. I believe that it is a requirement. ARIA in HTML says:

> [WAI-ARIA](https://www.w3.org/TR/html-aria/#bib-wai-aria-1.2) identifies roles which have [prohibited states and properties](https://www.w3.org/TR/wai-aria-1.2/#prohibitedattributes). These roles do not allow certain WAI-ARIA attributes to be specified by authors. HTML elements which expose these implicit WAI-ARIA roles also prohibit authors from specifying these WAI-ARIA attributes.

And then it has the following next to the `input type="checkbox" in the _Rules of ARIA attribute usage by HTML element_ table:

> Authors [MUST NOT use the aria-checked attribute on input type=checkbox elements](https://www.w3.org/TR/html-aria/#att-checked).

... and the following next to "Any element where the [checked](https://html.spec.whatwg.org/multipage/input.html#attr-input-checked) attribute is allowed" in the _Rules of ARIA attribute usage by HTML feature_ table:

> Authors MUST NOT use the [aria-checked](https://www.w3.org/TR/wai-aria-1.2/#aria-checked) attribute on any element where the [checkedness](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-fe-checked), or the indeterminate checked value of the element can be in opposition to the current value of the aria-checked attribute.

It causes an issue where the appropriate state may not be exposed to the user which in my view may be a failure of 4.1.2. As the rule's expectation number one references ARIA in HTML I think that it is valid.

Open to any suggestions. Do you still feel strongly that this should be a separate rule or should I just make it clearer in the background section and reference ARIA in HTML?

Reply:
Good points 👍 
I agree that ARIA in HTML makes this an author requirement 👍 
And indeed, the rule seems to already include ARIA in HTML stuff 👍 
(this does raise the question of the requirement mapping only going to WAI-ARIA and some ARIA technique, but that's a bit unrelated).

I think the background note should also refer to ARIA in HTML (the existing one does so already).

Reply:
Hey @Jym77 
Please review again as per my and Wilco's conversation. Thanks!